homeassistant/components/shelly/switch.py

```diff
@@ -243,12 +202,21 @@ def _update_callback(self) -> None:
         super()._update_callback()
 
 
-class BlockRelaySwitch(ShellyBlockEntity, SwitchEntity):
+class BlockRelaySwitch(ShellyBlockAttributeEntity, SwitchEntity):
     """Entity that controls a relay on Block based Shelly devices."""
 
-    def __init__(self, coordinator: ShellyBlockCoordinator, block: Block) -> None:
+    entity_description: BlockSwitchDescription
+
+    def __init__(
+        self,
+        coordinator: ShellyBlockCoordinator,
+        block: Block,
+        attribute: str,
+        description: BlockSwitchDescription,
+    ) -> None:
         """Initialize relay switch."""
-        super().__init__(coordinator, block)
+        super().__init__(coordinator, block, attribute, description)
+        self._attr_unique_id: str = f"{super().unique_id}"
```

Comment:
```suggestion
```

I'm not sure this is needed since you are setting it to exactly what the parent classes sets it to